eyeshade/migrations/0012_balance_snapshots/up.sql

```diff
@@ -0,0 +1,26 @@
+select execute($$
+insert into migrations (id, description) values ('0012', 'balance_snapshots');
+
+create table payout_reports(
+  id                    uuid primary key,
+  created_at            timestamp with time zone not null default current_timestamp,
+  updated_at            timestamp with time zone not null default current_timestamp,
+  latest_transaction_at timestamp with time zone,
+  completed             boolean default false
+);
+
+create table balance_snapshots(
+  id           uuid primary key default uuid_generate_v4(),
+  created_at   timestamp with time zone not null default current_timestamp,
+  snapshot_id  uuid not null references payout_reports(id) on delete cascade,
+  account_id   text not null,
+  account_type text not null,
+  balance      numeric(28, 18) not null default 0.0
```

Comment:
I don't think this is hugely significant but I think we may not want to default to 0.0 here, since we want to be sure that whatever value is saved as balance reflects exactly what's in their account